src/ngraph/pass/memory_layout.hpp

```diff
@@ -35,11 +35,12 @@ namespace ngraph
 class ngraph::pass::MemoryLayout : public FunctionPass
 {
 public:
-    MemoryLayout(size_t alignment = 1);
+    MemoryLayout(size_t alignment = 1, bool free_temps = true);
```

Comment:
A description of what free_temps does would be good. How about naming it something like `disable_memory_sharing` and have it default to false. If you named it disable_memory_sharing then I would not really need a comment.